common-auth/SECURITY.md

```diff
@@ -0,0 +1,65 @@
+
+```java
+List<ServletFilterConfiguration> servletFilterConfigurations = registryConfiguration.getServletFilters();
+if (servletFilterConfigurations != null && !servletFilterConfigurations.isEmpty()) {
+    for (ServletFilterConfiguration servletFilterConfiguration: servletFilterConfigurations) {
+        try {
+            FilterRegistration.Dynamic dynamic = environment.servlets().addFilter(servletFilterConfiguration.getClassName(), (Class<? extends Filter>)
+            Class.forName(servletFilterConfiguration.getClassName()));
+            dynamic.setInitParameters(servletFilterConfiguration.getParams());
+            dynamic.addMappingForUrlPatterns(EnumSet.allOf(DispatcherType.class), true, "/*");
+        } catch (Exception e) {
+            LOG.error("Error registering servlet filter {}", servletFilterConfiguration);
+            throw new RuntimeException(e);
+        }
+    }
+}
+```
+
+ServletFilterConfiguration is a java object representing any servlet filter used in schema registry. However the general idea is that one needs to
+add com.hortonworks.registries.auth.server.AuthenticationFilter for enabling authentication 
+
+The configuration for that filter in yaml is represented in params property as follows.
+
+```yaml
+servletFilters:
+ - className: "com.hortonworks.registries.auth.server.AuthenticationFilter"
+   params:
+     type: "kerberos"
+     kerberos.principal: "HTTP/web-service-host.com"
+     kerberos.keytab: "/path/to/keytab"
+     kerberos.name.rules: "RULE:[2:$1@$0]([jt]t@.*EXAMPLE.COM)s/.*/$MAPRED_USER/ RULE:[2:$1@$0]([nd]n@.*EXAMPLE.COM)s/.*/$HDFS_USER/DEFAULT"
+     token.validity: 36000
+```
+
+The servlet filter uses the principal `HTTP/{hostname}` (here hostname should be the host where the web-service runs) to login. Make sure that principal is 
+created as part of Kerberos setup
+
+Once configured, you must do `kinit` on client side using the principal of the user before accessing the web-service via the browser or some other client. 
+This principal also needs to be created first during Kerberos setup
+
+Here's an example of accessing web-service after the setup above:
+```bash
+curl  -i --negotiate -u:anyUser  -b ~/cookiejar.txt -c ~/cookiejar.txt  http://<web-service-host>:<port>/api/v1/
+```
+
+1. Firefox: Go to `about:config` and search for `network.negotiate-auth.trusted-uris` double-click to add value "http://<web-service-host>:<port>"
+2. Google-chrome: start from command line with: 
+   `google-chrome --auth-server-whitelist="*web-service-hostname" --auth-negotiate-delegate-whitelist="*web-service-hostname"`
+3. IE: Configure trusted websites to include "web-service-hostname" and allow negotiation for that website
+
+**Caution**: In AD MIT Kerberos setup, the key size is bigger than the default UI jetty server request header size. If using that, make sure you set 
+header buffer bytes to 65536
```

Comment:
the text implies that the header buffer is to be set in the jetty server, which according to what you say is not the case. The text should be clear